web/docker/entrypoint.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+if [ "$(id -u)" = '0' ]; then
+  # Set PostgreSQL password file from secrets.
+  pgpass=/run/secrets/pgpass
+  if [ -f $pgpass ]; then
+    cp $pgpass /.pgpass
+    chmod 0600 /.pgpass
+    chown codechecker:codechecker /.pgpass
+    export PGPASSFILE=/.pgpass
+  fi
+
+  # Change the owner of the workspace directory
+  mkdir -p /workspace
+  chown codechecker:codechecker /workspace
+
+  # Execute this script again with codechecker user.
+  exec gosu codechecker "$0" "$@"
+fi
+
+exec "$@"
```

Comment:
Does this script support catching signals, foreground execution, background execution, etc. well? So far I've seen significant use of [`tini`](https://github.com/krallin/tini) as Docker entrypoints.